server/service/src/document/mod.rs

```diff
@@ -1,28 +1,24 @@
+use chrono::{DateTime, Utc};
 use repository::RepositoryError;
 
 use crate::service_provider::{ServiceContext, ServiceProvider};
 
-use self::raw_document::RawDocument;
-
 pub mod document_registry;
 pub mod document_service;
 pub mod form_schema_service;
 pub mod raw_document;
 
-/// Checks that there is no document with a later timestamp
+/// Checks if the doc is the latest in the DB
 pub(crate) fn is_latest_doc(
     ctx: &ServiceContext,
     service_provider: &ServiceProvider,
-    doc: &RawDocument,
+    doc_name: &str,
+    doc_timestamp: DateTime<Utc>,
 ) -> Result<bool, RepositoryError> {
-    let latest_existing = service_provider
+    let Some(latest_existing) = service_provider
         .document_service
-        .document(ctx, &doc.name, None)?;
-    if let Some(lastest_existing) = latest_existing {
-        if lastest_existing.timestamp > doc.timestamp {
-            // newer doc already exist
-            return Ok(false);
-        }
-    }
-    Ok(true)
+        .document(ctx, doc_name, None)? else {
+        return Ok(true);
+    };
+    Ok(latest_existing.datetime <= doc_timestamp)
```

Comment:
I think explicit sort would help (to guarantee that this wont' break if default sort is ever changed), would also help other validating the logic, I had to look up: https://github.com/openmsupply/open-msupply/blob/d8860428d38e45c382ab57a665c61801fdc7ece5/server/repository/src/db_diesel/document.rs#L271-L272